Event/CMakeLists.txt

```diff
@@ -12,40 +21,56 @@ string(REGEX REPLACE "\n$" "" GIT_SHA1 "${GIT_SHA1}")
 # Copies the file 'Version.h.in', substitutes the value of GIT_SHA1 and
 # writes it out to Version.h.  
 configure_file(
-    ${CMAKE_CURRENT_SOURCE_DIR}/include/Event/Version.h.in
-    ${CMAKE_CURRENT_BINARY_DIR}/include/Event/Version.h
+    ${PROJECT_SOURCE_DIR}/include/Event/Version.h.in
+    ${PROJECT_SOURCE_DIR}/include/Event/Version.h
 )
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/include/Event/Version.h DESTINATION ${CMAKE_INSTALL_PREFIX}/include/Event/)
+#install(FILES ${PROJECT_SOURCE_DIR}/include/Event/Version.h DESTINATION ${CMAKE_INSTALL_PREFIX}/include/Event/)
+
+# Search and configure ROOT
+find_package(ROOT 6.16 CONFIG REQUIRED)
+
+# Generate the ROOT dictionary.  The following allows the use of the macro 
+# used to generate the dictionary. 
+include("${ROOT_DIR}/RootMacros.cmake")
+
+# Unfortunately, the Event headers need to be included globably for the 
+# dictionary to be generated correctly. 
+include_directories(${PROJECT_SOURCE_DIR}/include)
+
+# Generate the ROOT dictionary
+root_generate_dictionary(EventDic ${PROJECT_SOURCE_DIR}/include/Event/EventDef.h LINKDEF ${PROJECT_SOURCE_DIR}/include/Event/EventLinkDef.h)
```

Comment:
The generated root dictionary references the files that you input here, so you should input the installed files. That way, we can delete the source and build files and still use the installed event dictionary.

Reply:
Is there a reason why you would delete the src files? Is that what you are doing in the container?

Reply:
Yes, that allows us to remove the source and build files before compressing the container. So it isn't necessary, just helpful.